testsuite/pytests/test_stdp_synapse.py

```diff
@@ -348,11 +412,20 @@ def plot_weight_evolution(
         plt.close(fig)
 
     def test_stdp_synapse(self):
-        self.dendritic_delay = float("nan")
         self.init_params()
-        for self.dendritic_delay in [1.0, self.resolution]:
-            self.init_params()
-            for self.nest_neuron_model in ["iaf_psc_exp", "iaf_cond_exp"]:
-                fname_snip = "_[nest_neuron_mdl=" + self.nest_neuron_model + "]"
-                fname_snip += "_[dend_delay=" + str(self.dendritic_delay) + "]"
-                self.do_nest_simulation_and_compare_to_reproduced_weight(fname_snip=fname_snip)
+        for self.dendritic_delay in (1.0, 0.5, self.resolution):
+            self.synapse_parameters["delay"] = self.dendritic_delay
+            for self.min_delay in (1.0, 0.4, self.resolution):
+                for self.max_delay in (3.0, 1.0):
+                    self.min_delay = min(self.min_delay, self.max_delay)
+                    self.max_delay = max(self.min_delay, self.max_delay)
+                    for self.nest_neuron_model in ("iaf_psc_exp", "iaf_cond_exp"):
+                        for self.neuron_parameters["t_ref"] in (self.resolution, 0.5, 1.0, 1.1, 2.5):
+                            fname_snip = "_[nest_neuron_mdl=" + self.nest_neuron_model + "]"
+                            fname_snip += "_[dend_delay=" + str(self.dendritic_delay) + "]"
+                            fname_snip += "_[t_ref=" + str(self.neuron_parameters["t_ref"]) + "]"
+                            self.do_nest_simulation_and_compare_to_reproduced_weight(fname_snip=fname_snip)
+
+
+if __name__ == "__main__":
```

Comment:
remove

Reply:
Why? Is it bad to make the test runnable by itself? Or because you should always run it using pytest?

Reply:
Yeah, the idea is that the tests should always be run through pytest.